core/src/main/scala/cromwell/core/PathFactory.scala

```diff
@@ -4,20 +4,38 @@ import java.io.Writer
 import java.nio.file.{FileSystem, Path}
 
 import scala.collection.immutable.Queue
-import scala.util.{Success, Try}
+import scala.util.{Success, Failure, Try}
+
+class FileSystemNotFound(str: String) extends CromwellFatalException(
+  new IllegalArgumentException(s"Could not find suitable filesystem to parse $str")
+)
 
 trait PathFactory {
+  private val schemeMatcher = """([a-z]*://).*""".r
```

Comment:
you probably want `+` and not `*` for at least the first quantifier, maybe the second